flake: TestAPI/Error/DuringInjection

```
panic: test timed out after 20m0s
	running tests:
		TestAPI/Error/DuringInjection (20m0s)

goroutine 575 [running]:
testing.(*M).startAlarm.func1()
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:2484 +0x308
created by time.goFunc
	/Users/runner/work/_tool/go/1.24.6/arm64/src/time/sleep.go:215 +0x38

goroutine 1 [chan receive, 19 minutes]:
testing.tRunner.func1()
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1753 +0x43c
testing.tRunner(0x1400054e1c0, 0x140003b5c68)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1798 +0x120
testing.runTests(0x140005417b8, {0x102fc9140, 0x14, 0x14}, {0x1400001f5c0?, 0x7?, 0x1030d0380?})
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:2277 +0x3ec
testing.(*M).Run(0x1400023d400)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:2142 +0x588
main.main()
	_testmain.go:85 +0x90

goroutine 6 [sync.Cond.Wait, 20 minutes]:
sync.runtime_notifyListWait(0x140005b8950, 0x0)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/runtime/sema.go:597 +0x150
sync.(*Cond).Wait(0x140005b8940)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/sync/cond.go:71 +0xc4
github.com/cihub/seelog.(*asyncLoopLogger).processItem(0x140005ba1b0)
	/Users/runner/go/pkg/mod/github.com/cihub/seelog@v0.0.0-20170130134532-f561c5e57575/behavior_asynclooplogger.go:50 +0x80
github.com/cihub/seelog.(*asyncLoopLogger).processQueue(0x140005ba1b0)
	/Users/runner/go/pkg/mod/github.com/cihub/seelog@v0.0.0-20170130134532-f561c5e57575/behavior_asynclooplogger.go:63 +0x38
created by github.com/cihub/seelog.NewAsyncLoopLogger in goroutine 1
	/Users/runner/go/pkg/mod/github.com/cihub/seelog@v0.0.0-20170130134532-f561c5e57575/behavior_asynclooplogger.go:40 +0xc8

goroutine 7 [sync.Cond.Wait, 20 minutes]:
sync.runtime_notifyListWait(0x140005b8ad0, 0x0)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/runtime/sema.go:597 +0x150
sync.(*Cond).Wait(0x140005b8ac0)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/sync/cond.go:71 +0xc4
github.com/cihub/seelog.(*asyncLoopLogger).processItem(0x140005ba2d0)
	/Users/runner/go/pkg/mod/github.com/cihub/seelog@v0.0.0-20170130134532-f561c5e57575/behavior_asynclooplogger.go:50 +0x80
github.com/cihub/seelog.(*asyncLoopLogger).processQueue(0x140005ba2d0)
	/Users/runner/go/pkg/mod/github.com/cihub/seelog@v0.0.0-20170130134532-f561c5e57575/behavior_asynclooplogger.go:63 +0x38
created by github.com/cihub/seelog.NewAsyncLoopLogger in goroutine 1
	/Users/runner/go/pkg/mod/github.com/cihub/seelog@v0.0.0-20170130134532-f561c5e57575/behavior_asynclooplogger.go:40 +0xc8

goroutine 679 [chan receive, 19 minutes]:
github.com/coder/coder/v2/agent/agentcontainers.(*API).updaterLoop.func1()
	/Users/runner/work/coder/coder/agent/agentcontainers/api.go:671 +0xf8
github.com/coder/quartz.(*mockTickerFunc).fire(0x1400081d560, {0x140007c2fc0?, 0x1?, 0x100927dd4?})
	/Users/runner/go/pkg/mod/github.com/coder/quartz@v0.2.1/mock.go:522 +0x204
github.com/coder/quartz.(*Mock).advanceLocked.func1()
	/Users/runner/go/pkg/mod/github.com/coder/quartz@v0.2.1/mock.go:299 +0x3c
created by github.com/coder/quartz.(*Mock).advanceLocked in goroutine 678
	/Users/runner/go/pkg/mod/github.com/coder/quartz@v0.2.1/mock.go:298 +0x98

goroutine 33 [chan receive, 19 minutes]:
testing.tRunner.func1()
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1753 +0x43c
testing.tRunner(0x1400054f6c0, 0x1022fb840)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1798 +0x120
created by testing.(*T).Run in goroutine 1
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1851 +0x374

goroutine 364 [select, 19 minutes]:
github.com/coder/websocket.(*Conn).timeoutLoop(0x140000e2b40)
	/Users/runner/go/pkg/mod/github.com/coder/websocket@v1.8.13/conn.go:186 +0x130
created by github.com/coder/websocket.newConn in goroutine 132
	/Users/runner/go/pkg/mod/github.com/coder/websocket@v1.8.13/conn.go:148 +0x5d0

goroutine 678 [sync.WaitGroup.Wait, 19 minutes]:
sync.runtime_SemacquireWaitGroup(0x14000569c00?)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/runtime/sema.go:110 +0x2c
sync.(*WaitGroup).Wait(0x14000d19070)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/sync/waitgroup.go:118 +0x70
github.com/coder/quartz.(*Mock).advanceLocked(0x140004e1a40, {{0x102344d80?, 0x14000951500?}, 0x14000c07110?})
	/Users/runner/go/pkg/mod/github.com/coder/quartz@v0.2.1/mock.go:307 +0x1b8
created by github.com/coder/quartz.(*Mock).AdvanceNext in goroutine 517
	/Users/runner/go/pkg/mod/github.com/coder/quartz@v0.2.1/mock.go:369 +0x2f8

goroutine 165 [chan receive, 19 minutes]:
testing.tRunner.func1()
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1753 +0x43c
testing.tRunner(0x140008236c0, 0x1022fb9a0)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1798 +0x120
created by testing.(*T).Run in goroutine 33
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1851 +0x374

goroutine 660 [sync.Cond.Wait, 19 minutes]:
sync.runtime_notifyListWait(0x14000e27350, 0x0)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/runtime/sema.go:597 +0x150
sync.(*Cond).Wait(0x14000e27340)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/sync/cond.go:71 +0xc4
github.com/coder/quartz.(*mockTickerFunc).Wait(0x1400081d560, {0x14000af99e0?, 0x14000c4bb58?, 0x1009097dc?})
	/Users/runner/go/pkg/mod/github.com/coder/quartz@v0.2.1/mock.go:559 +0x13c
github.com/coder/coder/v2/agent/agentcontainers.(*API).updaterLoop.func2()
	/Users/runner/work/coder/coder/agent/agentcontainers/api.go:692 +0x90
runtime.Goexit()
	/Users/runner/work/_tool/go/1.24.6/arm64/src/runtime/panic.go:636 +0x60
testing.(*common).FailNow(0x14000951500)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1041 +0x48
testing.(*common).Fatalf(0x14000951500, {0x1018facd8?, 0x4?}, {0x140006c3270?, 0x102176da0?, 0x14000782048?})
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1125 +0x64
go.uber.org/mock/gomock.(*Controller).Call.func1(0x14000e6ed20, {0x102176da0, 0x14000782048}, {0x101869755, 0x4}, {0x14000e27480, 0x4, 0x4})
	/Users/runner/go/pkg/mod/go.uber.org/mock@v0.5.0/gomock/controller.go:214 +0x32c
go.uber.org/mock/gomock.(*Controller).Call(0x14000e6ed20, {0x102176da0, 0x14000782048}, {0x101869755, 0x4}, {0x14000e27480, 0x4, 0x4})
	/Users/runner/go/pkg/mod/go.uber.org/mock@v0.5.0/gomock/controller.go:230 +0x6c
github.com/coder/coder/v2/agent/agentcontainers/acmock.(*MockContainerCLI).Copy(0x14000782048, {0x102321660, 0x14000475c70}, {0x1018a94e1, 0x11}, {0x14000c071f0, 0x66}, {0x1018b10f7, 0x13})
	/Users/runner/work/coder/coder/agent/agentcontainers/acmock/acmock.go:48 +0x170
github.com/coder/coder/v2/agent/agentcontainers.(*API).maybeInjectSubAgentIntoContainerLocked(0x140004a4b08, {0x1023215f0, 0x1400060ad70}, {{0xa4, 0xd6, 0x64, 0x2, 0x30, 0xb2, 0x46, ...}, ...})
	/Users/runner/work/coder/coder/agent/agentcontainers/api.go:1799 +0x18e4
github.com/coder/coder/v2/agent/agentcontainers.(*API).processUpdatedContainersLocked(0x140004a4b08, {0x1023215f0, 0x1400060ad70}, {{0x0, 0x0, 0x0}, {0x1400067d050, 0x1, 0x1}, {0x0, ...}})
	/Users/runner/work/coder/coder/agent/agentcontainers/api.go:1035 +0x1968
github.com/coder/coder/v2/agent/agentcontainers.(*API).updateContainers(0x140004a4b08, {0x1023215f0, 0x1400060ad70})
	/Users/runner/work/coder/coder/agent/agentcontainers/api.go:879 +0x23c
github.com/coder/coder/v2/agent/agentcontainers.(*API).updaterLoop(0x140004a4b08)
	/Users/runner/work/coder/coder/agent/agentcontainers/api.go:704 +0x760
created by github.com/coder/coder/v2/agent/agentcontainers.(*API).Start in goroutine 517
	/Users/runner/work/coder/coder/agent/agentcontainers/api.go:424 +0x1fc

goroutine 677 [sync.Cond.Wait, 19 minutes]:
sync.runtime_notifyListWait(0x14000e27350, 0x1)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/runtime/sema.go:597 +0x150
sync.(*Cond).Wait(0x14000e27340)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/sync/cond.go:71 +0xc4
github.com/coder/quartz.(*mockTickerFunc).waitForCtx(0x1400081d560)
	/Users/runner/go/pkg/mod/github.com/coder/quartz@v0.2.1/mock.go:547 +0xdc
created by github.com/coder/quartz.(*Mock).TickerFunc in goroutine 660
	/Users/runner/go/pkg/mod/github.com/coder/quartz@v0.2.1/mock.go:53 +0x344

goroutine 517 [chan receive, 19 minutes]:
github.com/coder/coder/v2/agent/agentcontainers.(*API).Close(0x140004a4b08)
	/Users/runner/work/coder/coder/agent/agentcontainers/api.go:2015 +0x71c
github.com/coder/coder/v2/agent/agentcontainers_test.TestAPI.func10.2.1()
	/Users/runner/work/coder/coder/agent/agentcontainers/api_test.go:2126 +0x34
runtime.Goexit()
	/Users/runner/work/_tool/go/1.24.6/arm64/src/runtime/panic.go:636 +0x60
testing.(*common).FailNow(0x14000951500)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1041 +0x48
testing.(*common).Fatalf(0x14000951500, {0x101908c19?, 0x2540be400?}, {0x14000d06000?, 0x2540be400?, 0x102344d01?})
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1125 +0x64
github.com/coder/quartz.AdvanceWaiter.MustWait({{0x102344d80?, 0x14000951500?}, 0x14000c07110?}, {0x102321660, 0x140000e9340})
	/Users/runner/go/pkg/mod/github.com/coder/quartz@v0.2.1/mock.go:247 +0x114
github.com/coder/coder/v2/agent/agentcontainers_test.TestAPI.func10.2(0x14000951500)
	/Users/runner/work/coder/coder/agent/agentcontainers/api_test.go:2161 +0x114c
testing.tRunner(0x14000951500, 0x1022fb9b8)
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1792 +0xe4
created by testing.(*T).Run in goroutine 165
	/Users/runner/work/_tool/go/1.24.6/arm64/src/testing/testing.go:1851 +0x374
```

seen here: https://github.com/coder/coder/actions/runs/16982738814/job/48145935854

--- a/agent/agentcontainers/api_test.go
+++ b/agent/agentcontainers/api_test.go
@@ -2096,9 +2096,6 @@ func TestAPI(t *testing.T) {
 				}
 			)
 
-			coderBin, err := os.Executable()
-			require.NoError(t, err)
-
 			// Mock the `List` function to always return the test container.
 			mCCLI.EXPECT().List(gomock.Any()).Return(codersdk.WorkspaceAgentListContainersResponse{
 				Containers: []codersdk.WorkspaceAgentContainer{testContainer},
@@ -2139,7 +2136,7 @@ func TestAPI(t *testing.T) {
 			require.Equal(t, http.StatusOK, rec.Code)
 
 			var response codersdk.WorkspaceAgentListContainersResponse
-			err = json.NewDecoder(rec.Body).Decode(&response)
+			err := json.NewDecoder(rec.Body).Decode(&response)
 			require.NoError(t, err)
 
 			// Then: We expect that there will be an error associated with the devcontainer.
@@ -2149,7 +2146,7 @@ func TestAPI(t *testing.T) {
 			gomock.InOrder(
 				mCCLI.EXPECT().DetectArchitecture(gomock.Any(), testContainer.ID).Return(runtime.GOARCH, nil),
 				mCCLI.EXPECT().ExecAs(gomock.Any(), testContainer.ID, "root", "mkdir", "-p", "/.coder-agent").Return(nil, nil),
-				mCCLI.EXPECT().Copy(gomock.Any(), testContainer.ID, coderBin, "/.coder-agent/coder").Return(nil),
+				mCCLI.EXPECT().Copy(gomock.Any(), testContainer.ID, gomock.Any(), "/.coder-agent/coder").Return(nil),
 				mCCLI.EXPECT().ExecAs(gomock.Any(), testContainer.ID, "root", "chmod", "0755", "/.coder-agent", "/.coder-agent/coder").Return(nil, nil),
 				mCCLI.EXPECT().ExecAs(gomock.Any(), testContainer.ID, "root", "/bin/sh", "-c", "chown $(id -u):$(id -g) /.coder-agent/coder").Return(nil, nil),
 			)
@@ -2157,8 +2154,8 @@ func TestAPI(t *testing.T) {
 			// Given: We allow creation to succeed.
 			testutil.RequireSend(ctx, t, fSAC.createErrC, nil)
 
-			_, aw := mClock.AdvanceNext()
-			aw.MustWait(ctx)
+			err = api.RefreshContainers(ctx)
+			require.NoError(t, err)
 
 			req = httptest.NewRequest(http.MethodGet, "/", nil)
 			rec = httptest.NewRecorder()

